### PR TITLE
Optimize getting payment reports from adpay

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: app/Client/GuzzleAdPayClient.php [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\RequestOptions;
+use Symfony\Component\HttpFoundation\Response;
+
+class GuzzleAdPayClient implements AdPay
+{
+    private const URI_CAMPAIGNS = '/api/v1/campaigns';
+
+    private const URI_PAYMENTS_TEMPLATE = '/api/v1/payments/%s';
```
Comment (m-pilarczyk) [inline app/Client/GuzzleAdPayClient.php]:
```suggestion
    private const URI_PAYMENTS_TEMPLATE = '/api/v1/payments/%d';
```

File: app/Client/GuzzleAdPayClient.php [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                $exception
+            );
+        }
+    }
+
+    public function getPayments(
+        int $timestamp,
+        bool $recalculate = false,
+        bool $force = false,
+        int $limit = PHP_INT_MAX,
```
Comment (m-pilarczyk) [inline app/Client/GuzzleAdPayClient.php]:
```suggestion
        ?int $limit = null,
```

File: app/Client/GuzzleAdPayClient.php [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        int $offset = 0
```
Comment (m-pilarczyk) [inline app/Client/GuzzleAdPayClient.php]:
```suggestion
        ?int $offset = null
```

**Review by m-pilarczyk — CHANGES_REQUESTED** [review]

File: app/Client/GuzzleAdPayClient.php (hunk 1 above)
Comment (PawelPodkalicki) [reply app/Client/GuzzleAdPayClient.php]: changed

File: app/Client/GuzzleAdPayClient.php (hunk 2 above)
Comment (PawelPodkalicki) [reply app/Client/GuzzleAdPayClient.php]: changed

File: app/Client/GuzzleAdPayClient.php (hunk 3 above)
Comment (PawelPodkalicki) [reply app/Client/GuzzleAdPayClient.php]: changed

**Review by m-pilarczyk — DISMISSED** [review]

**Pull request merged**